src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -0,0 +1,302 @@
+
+import com.intellij.codeInsight.intention.IntentionAction;
+import com.intellij.lang.ASTNode;
+import com.intellij.lang.annotation.Annotation;
+import com.intellij.lang.annotation.AnnotationHolder;
+import com.intellij.lang.annotation.Annotator;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.projectRoots.Sdk;
+import com.intellij.openapi.roots.ModuleRootManager;
+import com.intellij.openapi.roots.OrderRootType;
+import com.intellij.openapi.roots.ProjectRootManager;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.plugins.haxe.lang.lexer.HaxeTokenTypes;
+import com.intellij.plugins.haxe.lang.psi.*;
+import com.intellij.plugins.haxe.util.HaxeClassModel;
+import com.intellij.plugins.haxe.util.HaxeMethodModel;
+import com.intellij.plugins.haxe.util.HaxePsiUtils;
+import com.intellij.plugins.haxe.util.HaxeResolveUtil;
+import com.intellij.psi.*;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.psi.util.PsiUtil;
+import com.intellij.util.IncorrectOperationException;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+public class HaxeSemanticAnnotator implements Annotator {
+
+
+  @Override
+  public void annotate(PsiElement element, AnnotationHolder holder) {
+    PsiFile file = element.getContainingFile();
+    // Analyze upon file changes
+    Long analyzedFileTime = file.getUserData(HaxeSemanticError.KEY_ANALYZED_METHOD_TIME);
+    if (analyzedFileTime == null || analyzedFileTime != file.getModificationStamp()) {
+      analyze(file);
+      file.putUserData(HaxeSemanticError.KEY_ANALYZED_METHOD_TIME, file.getModificationStamp());
+    }
+
+    List<HaxeSemanticError> errors = element.getUserData(HaxeSemanticError.KEY_HAXE_ERROR);
+    if (errors != null) {
+      for (HaxeSemanticError error : errors) {
+        Annotation annotation = holder.createErrorAnnotation(element, error.message);
+        if (error.quickfix != null) annotation.registerFix(error.quickfix);
+      }
+    }
+  }
+
+
+  static void analyze(final PsiElement element) {
+    HaxeSemanticError.clearErrors(element);
+
+    if (element instanceof PsiJavaToken) {
+      if (((PsiJavaToken)element).getTokenType() == HaxeTokenTypes.LITINT) {
+        //element.putUserData(KEY_HAXE_ERROR, "litint!");
+      }
+    }
+    else if (element instanceof HaxePackageStatement) {
+      PackageChecker.check((HaxePackageStatement)element);
+    }
+    else if (element instanceof HaxeMethod) {
+      HaxeClass clazz = HaxePsiUtils.getAncestor(element, HaxeClass.class);
```

Comment:
To that search for ancestors and not just its parent? I will check and change it. Thanks!